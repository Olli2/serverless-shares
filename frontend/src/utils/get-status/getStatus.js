export const getStatus = (value) => {
    if (value > 0) {
        return 'positive';
    } else if (value === 0) {
        return 'neutral';
    }
    return 'negative';
}