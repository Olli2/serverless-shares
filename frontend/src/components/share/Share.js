import React from 'react';
import { getStatus } from '../../utils/get-status/getStatus';
import { isNil } from 'lodash';

export const Share = ({share}) => {
    return isNil(share) ? (
        <></>
    ) : (
        <div className={`share ${getStatus(share.performance)}`}>
            <p className="share__title" >{share.name}</p>
            <p className="share__info">{share.performance}</p>
        </div>
    )
}