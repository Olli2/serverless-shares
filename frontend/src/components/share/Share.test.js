import React from 'react';
import {render} from '@testing-library/react';
import {Share} from './Share';
import { getStatus } from '../../utils/get-status/getStatus'

jest.mock('../../utils/get-status/getStatus', () => ({
    getStatus: jest.fn() 
}))

describe('Share component', () => {
    afterEach(() => {
        jest.clearAllMocks()
    })

    it('should match with snapshot', () => {
        const actual = render(<Share share={{}}/>)
        expect(actual.asFragment).toMatchSnapshot()
    })

    it('should call getStatus', () => {
        render(<Share share={{performance: 999}}/>)
        expect(getStatus).toHaveBeenCalledTimes(1)
        expect(getStatus).toHaveBeenCalledWith(999)
    })
})