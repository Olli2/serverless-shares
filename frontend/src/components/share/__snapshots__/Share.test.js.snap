// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share component should match with snapshot 1`] = `[Function]`;
